### PR TITLE
Added support for getting CFN output values in aws metadata plugin

### PR DESCRIPTION
**Pull request by kencochrane**
This allows you to gather values from the cloudformation outputs for a stack. This is helpful when trying to get results from a stack and using it as inputs down stream.

in a template you could do something like this.

```
{{ var "/my/cfn/output" (metadata "var/export/cfn/config/Outputs/MyStackOutput/OutputValue" )}}
```

Signed-off-by: Ken Cochrane <kencochrane@gmail.com>

### CODE REVIEW

File: docs/controller/ingress/example.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@
 #m4fghruao79i        t2                  replicated          1/1                 nginx:latest        *:7777->80/tcp
 #
 # Remove the service
-# docker service rm test1
+# docker service rm t2
```
Comment (kencochrane) [inline docs/controller/ingress/example.yml]: fixed typo

File: pkg/provider/aws/plugin/metadata/funcs.go [hunk 2]
```diff
@@ -135,9 +135,19 @@ func cfn(clients AWSClients, name string) (interface{}, error) {
...
 
+	// index outputs by name
+	outputs := map[string]interface{}{}
+	for _, p := range output.Stacks[0].Outputs {
+		if p.OutputKey == nil {
+			continue
+		}
+		outputs[*p.OutputKey] = p
+	}
+
```
Comment (kencochrane) [inline pkg/provider/aws/plugin/metadata/funcs.go]: I tried to test locally, but not sure how to test, any suggestions would be helpful.

**Comment by kencochrane** [comment]
It looks like the test failure isn't related to my change.

File: pkg/provider/aws/plugin/metadata/funcs.go (hunk 2 above)
Comment (chungers) [reply pkg/provider/aws/plugin/metadata/funcs.go]: The only way I can see is to mock the API call but tbh it really isn't testing much given how simple the code fragment is.

Comment (kencochrane) [reply pkg/provider/aws/plugin/metadata/funcs.go]: Thanks

**Comment by kencochrane** [comment]
I'm not sure why the test keep failing. It doesn't look like it is related to my change. I could be wrong.

**Pull request merged**